### PR TITLE
Kan 61 env zu laufzeit anpassen

### PR DESCRIPTION
**Pull request by SuffPanda**
close KAN-61

### CODE REVIEW

File: .github/workflows/pr-workflow.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
         uses: docker/setup-buildx-action@v1
   
       - name: Build Docker Image
-        run: docker build . --file ./Dockerfile
+        run: docker build . --file ./Dockerfile.local
```
Comment (mueld) [inline .github/workflows/pr-workflow.yml]: This workflow is intended to ensure that the current push is buildable. So replace the Dockerfile with the file you actually need to build the dev stage

File: .github/workflows/production-stage.yml [hunk 2]
```diff
@@ -63,14 +64,27 @@ jobs:
...
+          node-version: '20.11.1'
+      
+      - name: Install Minimal Dependencies
+        working-directory: ./.github
+        run: npm install
+        
+      - name: Switch to Production Env
+        working-directory: ./.github
+        run: npm run import-meta-env:production
+
```
Comment (mueld) [inline .github/workflows/production-stage.yml:79]: if you rebuild the container when you push it to production, you can't ensure that the build from the development phase will work in production. => "npm install" always installs the latest packages!

Comment (SuffPanda) [reply .github/workflows/production-stage.yml:79]: Here, `npm install` is used in a different context. I added a new [package.json](https://github.com/OlympGuide/Frontend/blob/c7def169e2a422517a8f5b7c249c3e5785f91ce9/.github/package.json) file in the /.github folder containing a minimal setup to replace the env variables. This does not rebuild the container but modifies solely one line in the index.html file that defines the env variables for the website.

File: src/components/OlympMap.vue [hunk 3]
```diff
@@ -101,9 +99,12 @@ const createIcons = (): IconKeyMap => {
   return iconKeyMap;
 };
 
-const loadSportFields = (): void => {
-  sportFieldStore.loadSportFields();
-  sportFields.value = sportFieldStore.sportFields;
+const loadSportFields = async (): Promise<void> => {
+  await sportFieldStore.loadSportFields()
+  .then(_ => {
```
Comment (aubrydario) [inline src/components/OlympMap.vue:104]: await + then? Wieso beides? Kann man das nicht einfach mit await lösen?

File: .github/package.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "minimal",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "import-meta-env:local": "import-meta-env -x ../.env.example.public -e ../.env.local -p ../dist/index.html",
+    "import-meta-env:development": "import-meta-env -x ../.env.example.public -e ../.env.development -p ../dist/index.html",
+    "import-meta-env:production": "import-meta-env -x ../.env.example.public -e ../.env.production -p ../dist/index.html"
```
Comment (aubrydario) [inline .github/package.json:9]: Sollten die '.env.example.public' heissen?

Comment (SuffPanda) [reply .github/package.json:9]: Yes, es gibt nebem dem .public auch ein .private (nicht implementiert hier). Der hat env variables, die nicht commitet werden dürfen (zB Tokens).

**Pull request merged**